### DISCUSSION TITLE
Token expiration refresh

### DISCUSSION THREAD

**Discussion by cocadepoma** [zitadel/zitadel#7107] [Q&A]
I am using a React app with the library oidc-client-ts and I cannot refresh the user's tokens.

I am quite new on OIDC and Zitaldel and I would like to refresh the tokens before expiration but I have no idea how to perform it.

I set a 0.05-hour expiration for the access_token and id_token, and 10 days for refresh_token expiration and I can login and logout in to the app, but I am not able to achieve the silent sign-in/login in an I-frame since apparently, Zitadel does not support it and I don't know how can I do that manually.

Can someone help me find a solution to refresh the tokens, please?

I am glad to help with React examples for improving the Zitadel documentation since I did not find any solution for this (if finally I get it working).

I appreciate any help you can provide.

**Comment by hifabienne** [comment]
hei @peintnermax 
Can you help here?
